backend/zfs_operations/services/dataset_service.py

```
@@ -0,0 +1,576 @@
+            if exists_result.value:
+                return Result.failure(DatasetAlreadyExistsError(str(name)))
+            
+            # Build create command
+            command_args = ["create"]
+            
+            # Add properties if provided
+            if properties:
+                validated_props = self._validator.validate_zfs_properties(properties)
+                if not validated_props:
+                    return Result.failure(DatasetException(
+                        f"Invalid properties specified",
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)